addons/point_of_sale/point_of_sale.py

```diff
@@ -590,16 +593,15 @@ def _process_order(self, cr, uid, order, context=None):
         if not float_is_zero(order['amount_return'], self.pool.get('decimal.precision').precision_get(cr, uid, 'Account')):
             cash_journal = session.cash_journal_id
             if not cash_journal:
-                cash_journal_ids = filter(lambda st: st.journal_id.type=='cash', session.statement_ids)
-                if not len(cash_journal_ids):
+                cash_journals = filter(lambda journal: journal.type=='cash', self.pool['account.journal'].browse(cr, uid, journal_ids, context=context))
```

Comment:
could be replaced by a search `[('type', '=', 'cash'), ('id', 'in', journal_ids)]`